lib/enigma.rb

```diff
@@ -43,6 +51,14 @@ def msg_converter(message, key = nil, date = nil)
     end
   end
 
+  def msg_converter_decrypt(message, key = nil, date = nil)
+    shift_parameters = @shift_infomation.shift(key, date)
+    message_in_groups = msg_isolator(message)
+    message_in_groups.map.with_index do |message_increment, index|
+      character_converter_decrypt(message_increment, shift_parameters.values)
```

Comment:
only difference from encryption version is that it calls on a new character_converter